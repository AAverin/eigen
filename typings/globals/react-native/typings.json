{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
