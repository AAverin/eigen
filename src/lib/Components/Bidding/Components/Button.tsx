import React from "react"
import { StyleSheet } from "react-native"

import PrimaryBlack, { PrimaryBlackProps } from "lib/Components/Buttons/PrimaryBlack"
import { ButtonProps, SecondaryOutlineButton } from "../../Buttons"
import { Flex } from "../Elements/Flex"

export class Button extends React.Component<PrimaryBlackProps> {
  render() {
    const { textStyle, ...props } = this.props

    return (
      <Flex height={50}>
        <PrimaryBlack textStyle={[textStyle]} {...props} />
      </Flex>
    )
  }
}

export class BidGhostButton extends React.Component<ButtonProps> {
  render() {
    const { style, textStyle, ...props } = this.props

    return <SecondaryOutlineButton style={[styles.default, style]} textStyle={[textStyle]} {...props} />
  }
}

const styles = StyleSheet.create({
  default: {
    height: 50,
  },
})
