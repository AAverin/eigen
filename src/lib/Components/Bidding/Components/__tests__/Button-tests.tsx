import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { Button } from "../Button"

import { BiddingThemeProvider } from "../BiddingThemeProvider"

it("renders properly", () => {
  const bg = renderer
    .create(
      <BiddingThemeProvider>
        <Button text="next" />
      </BiddingThemeProvider>
    )
    .toJSON()
  expect(bg).toMatchSnapshot()
})
