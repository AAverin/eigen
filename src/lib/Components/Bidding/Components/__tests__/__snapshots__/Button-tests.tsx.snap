// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<View
  height={50}
  style={
    Array [
      Object {
        "height": 50,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 1)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "color": "white",
              "fontSize": 14,
              "opacity": 1,
            },
            Object {
              "fontFamily": "AvantGardeGothicITC",
            },
          ]
        }
      >
        NEXT
      </Text>
    </View>
  </View>
</View>
`;
